openquake/hazardlib/gsim/bahrampouri_2021.py

```diff
@@ -74,26 +91,79 @@ def _get_site_term(ctx, C):
     fsite = C['c1'] * np.log(ctx.vs30)
     return fsite
 
-
 def _get_stddevs(C):
+
     sig = C['sig']
     tau = C['tau']
     phi = np.sqrt(C['phi_ss']**2 + C['phi_s2s']**2)
+    
     return sig, tau, phi
 
+def _check_cm_ck(kyushu, honshu, ctx):
```

Comment:
Is the correction for the volcanic front mandatory? Otherwise, we should have an option at the instantiation level to disable it.

Reply:
The correction for the volcanic front is not mandatory. In the absence of the functions `_check_cm_ck` and `_compute_volcanic_distances`, the path term  <img width="498" alt="Screenshot 2022-06-10 at 3 39 40 PM" src="https://user-images.githubusercontent.com/78085634/173043483-07456d08-78d9-4b58-bbb7-fcd8626bd939.png"> 
reduces to 
<img width="265" alt="Screenshot 2022-06-10 at 3 30 23 PM" src="https://user-images.githubusercontent.com/78085634/173041691-58754047-b3c5-401f-9cf1-b622a90286ce.png">

There are two applicability scenarios:
1. sites outside of Japan - The current implementation calculates cm=ck=rrup_b=rrup_f to be equal to 0 and the path term collapses to the same form as above. 
2. sites within Japan - the present implementation holds good. 
Hence, instantiation may not be necessary. Please let me know if I was not clear or didn't understand your input.

Reply:
Thanks. Wrt to point 1, is this case covered by the tests you created?

Reply:
yes. The tests covers case 1.

Reply:
thank you. LGTM